slack-java-client/src/main/java/com/hubspot/slack/client/SlackWebClient.java

```diff
@@ -930,24 +933,27 @@ protected String getInitialOffset() {
         .setUrl(config.getSlackApiBasePath().get() + "/" + method.getMethod());
   }
 
-  private <T extends SlackResponse> CompletableFuture<Result<T, SlackError>> executeLoggedAs(
+  @VisibleForTesting
+  <T extends SlackResponse> CompletableFuture<Result<T, SlackError>> executeLoggedAs(
       SlackMethod method,
       HttpRequest request,
       Class<T> responseType
   ) {
     long requestId = REQUEST_COUNTER.getAndIncrement();
-    return executeLogged(requestId, method, request)
+    requestDebugger.debug(requestId, method, request);
+
+    Stopwatch timer = Stopwatch.createStarted();
+    double acquireSeconds = acquirePermit(method);
+
+    if (acquireSeconds == RATE_LIMIT_SENTINEL_VALUE) {
+      responseDebugger.debugProactiveRateLimit(requestId, method, request);
+      return CompletableFuture.completedFuture(Result.err(SlackError.of(SlackErrorType.RATE_LIMITED.key())));
```

Comment:
I utilized the existing `SlackErrorType` rather than creating a new error because the rate limit result is intended to be the same whether it came from Slack or was calculated by a configured rate limiter.